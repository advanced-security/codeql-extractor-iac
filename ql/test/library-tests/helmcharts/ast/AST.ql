private import iac

query predicate helmchart(HelmChart::Document n) { any() }

query predicate helmchartValues(HelmChartValues::Document n) { any() }

query predicate helmchartAssignedValues(HelmChart::Document doc) { exists(doc.getValues()) }

query predicate helmchartSpec(HelmChart::Spec n) { any() }

query predicate helmchartContainers(HelmChart::Container n) { any() }

query predicate helmchartContainerSecurityContext(HelmChart::SecurityContext n) { any() }

query predicate helmchartDeps(HelmChart::Dependency n) { any() }

query predicate helmchartData(HelmChart::HelmData n) { any() }
