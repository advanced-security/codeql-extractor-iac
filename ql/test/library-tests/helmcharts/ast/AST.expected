helmchart
| Chart.yml:1:1:30:26 | HelmChart Document |
| random.yml:1:1:11:13 | HelmChart Document |
helmchartValues
| values.yaml:2:1:104:23 | HelmChart Value Document |
| values.yaml:4:3:10:27 | HelmChart Value Document |
| values.yaml:12:15:12:16 | HelmChart Value Document |
| values.yaml:17:3:104:23 | HelmChart Value Document |
| values.yaml:36:16:36:17 | HelmChart Value Document |
| values.yaml:38:11:38:12 | HelmChart Value Document |
| values.yaml:40:19:40:20 | HelmChart Value Document |
| values.yaml:42:14:42:15 | HelmChart Value Document |
| values.yaml:79:5:79:6 | HelmChart Value Document |
| values.yaml:95:5:95:6 | HelmChart Value Document |
helmchartAssignedValues
helmchartSpec
| Chart.yml:25:3:30:26 | securityContext: |
helmchartContainers
| Chart.yml:29:7:30:26 | Container |
helmchartContainerSecurityContext
| Chart.yml:26:5:28:2 | SecurityContext |
helmchartDeps
| Chart.yml:14:5:17:2 | nginx |
| Chart.yml:17:5:19:53 | memcached |
helmchartData
| Chart.yml:22:3:22:26 | HelmData |
